pandas/core/ops/__init__.py

```diff
@@ -311,6 +311,11 @@ def to_series(right):
             )
 
     elif is_list_like(right) and not isinstance(right, (ABCSeries, ABCDataFrame)):
+        # GH 36702. Raise when attempting arithmetic with list of array-like.
+        if any(is_array_like(el) for el in right):
```

Comment:
We could just check the first element for a speed boost, but I think consistency is more important than speed here. Especially considering that performing arithmetic on a DataFrame and a list isn't what users should be doing anyway.

Reply:
can you run the indexing / arithmetic benchmarks and see if any regressions

Reply:
No significant changes in arithmetic. Will run indexing tomorrow.